recirq/otoc/utils.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions used for the OTOC experiment."""
+
+import pickle
+from typing import Any, Optional, Tuple, Dict, List, Sequence
+
+import cirq
+import numpy as np
+from scipy.optimize import curve_fit
+
+
+def save_data(data: Any, file_name: str) -> None:
```

Comment:
pickle is a very bad format for data storage. It is an opaque format that can break very easily between python versions or even if you change the name of a class or function or even if you change a class attribute from a normal attribute to a `@property`. This is probably outside the scope of the current review, but we should open an issue to modify these routines to use the recirq data collection idioms and `recirq.save` and `recirq.load`

Reply:
Yeah I agree that pickle seems to be fickle (no pun intended) with version compatibility. I will open an issue.